src/intermediate/Instruction.cpp

```diff
@@ -421,6 +421,18 @@ bool IntermediateInstruction::replaceValue(const Value oldValue, const Value new
     return replaced;
 }
 
+bool IntermediateInstruction::isConstantInstruction() const
+{
+    if(dynamic_cast<const LoadImmediate*>(this) != nullptr)
+    {
+        return true;
+    }
+    auto& args = getArguments();
+    return std::all_of(args.begin(), args.end(), [](const Value& arg) { return !arg.isWriteable(); }) &&
```

Comment:
Couldn't you use precalculate() for this?

Reply:
It seems that `precalculate` can be used at here. What should I set `numIterations` as argument for this usage?

Reply:
That is basically how far back the calculation should look to find a constant result. So a value of 1 (or maybe 0, not sure right now) will only test the instruction itself to generate a constant result.
With a higher value, better optimization can be achieved, but it will take longer. So for now to be on the safe side, you could go with the minimum value to only look at the current instruction. This should have the same result as your current test.